src/emu/recording.cpp

```diff
@@ -178,8 +178,8 @@ bool avi_movie_recording::initialize(running_machine &machine, std::unique_ptr<e
 	// build up information about this new movie
 	avi_file::movie_info info;
 	info.video_format = 0;
-	info.video_timescale = 1000 * (screen() ? screen()->frame_period().as_hz() : screen_device::DEFAULT_FRAME_RATE);
-	info.video_sampletime = 1000;
+	info.video_timescale = screen() ? 1000000000 : screen_device::DEFAULT_FRAME_RATE;
+	info.video_sampletime = screen() ? screen()->frame_period().as_attoseconds() / 1000000000 : 1;
```

Comment:
Where does the one thousand million (or one milliard, or one billion, depending on how you define your powers of ten) constant come from?

You should really use `as_ticks` for doing this kind of conversion since `as_attoseconds` clamps to ±1 second so it will behave strangely in some cases, and you should use digit grouping in large numeric literals to make it easier to see order of magnitude.

What order of magnitude values are you going to get out of this?  Is there potential for it to overflow the 32-bit `video_sampletime` field?

Why are you using a different relationship between the two fields for systems with or without a screen?  Would it make more sense to always use the same timescale and use either the frame period from the screen or from the default frame rate to calculate the sample time?

Reply:
Dividing `ATTOSECONDS_PER_SECOND` by `frame_period()` would give the most accurate framerate value, but that requires 64-bit fields for numerator and denominator. So I shorten both values to make them fit into `int`s (as the AVI spec wants), and any info we lose by this truncation is not going to affect the resulting video, because it would have to be ridiculously long to accumulate enough error, and it would still not be distinguishable for a human.

Plain and obvious 60/1 for screenless machines felt like a preferred value, but I can also just switch to `1'000'000'000 / screen_device::DEFAULT_FRAME_RATE` for `video_sampletime` and make `video_timescale` 1'000'000'000 in all cases, even though it would not be exact 60 anymore but rather 60.0000024, and possibly confusing.

Reply:
Is there a smart way to choose a denominator to minimise the accumulated error, e.g. looking for a near factor of the frame interval?  If you’ve got freedom to choose both numerator and denominator to get both to fit within 32 bits, you should be able to do better than just using a fixed value of 1'000'000'000.